File: 540. Single Element in a Sorted Array/solution.js
/*
71 ms
44.8 MB
*/
/**
 * TC: O(n)
 * SC: O(1)
*/
/**
 * @author : SahilK-027
 * @brief : XOR
 */

/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNonDuplicate = function(nums) {
    let ans = 0;
    nums.forEach((i, idx)=>{
        ans ^= i;
    })
    return ans;
};;

File: 540. Single Element in a Sorted Array/solution.cpp
/*
49 ms
27.5 MB
*/
/**
 * TC: O(n)
 * SC: O(n)
*/
/**
 * @author : SahilK-027
 * @brief : map
 */


class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        unordered_map<int, int> mp;
        for(auto i: nums){
            mp[i]++;
        }
        for(auto i: mp){
            if(i.second == 1){
                return i.first;
            }
        }
        return -1;
    }
};